nautobot/extras/utils.py

```diff
@@ -207,6 +208,92 @@ def wrapper(model_class):
     return wrapper
 
 
+def update_registry_with_new_apps():
+    """
+    Updates the registry with new apps.
+
+    This function updates the registry model features for models.
+    """
+    from nautobot.extras.models import CustomField, Relationship, RelationshipAssociation, Role
+
+    lookup_confs = [
+        {
+            "model": CustomField,
+            "lookup_by": "field",
+            "field_names": ["_custom_field_data"],
+        },
+        {
+            "model": Relationship,
+            "lookup_by": "field",
+            "field_names": ["source_for_associations", "destination_for_associations"],
+            "extras": {"related_model": RelationshipAssociation},
+        },
+        {"model": Role, "lookup_by": "related_model", "related_model": Role},
```

Comment:
Wouldn't this just be equivalent to:

```suggestion
        {
            "model": Role,
            "lookup_by": "field",
            "field_names": ["role"],
            "extras": {"related_model": Role},
        },
```

removing the need for a separate `lookup_by_related_model` function?